xline/src/storage/index.rs

```diff
@@ -281,48 +272,52 @@ impl IndexOperate for Index {
         create_revision: i64,
         version: i64,
     ) {
-        let mut new_rev = KeyRevision::new(create_revision, version, revision, sub_revision);
-        new_rev.available = true;
         self.inner
-            .lock()
-            .index
-            .entry(key)
-            .or_insert_with(Vec::new)
-            .push(new_rev);
+            .get_or_insert(key, RwLock::new(Vec::new()))
+            .value()
+            .map_write(|mut revisions| {
+                revisions.push(KeyRevision::new(
+                    create_revision,
+                    version,
+                    revision,
+                    sub_revision,
+                ));
+            });
     }
 
     fn compact(&self, at_rev: i64) -> Vec<KeyRevision> {
```

Comment:
Is it possible that `insert` is called after `delete` that the old revisions may never be compacted.

Reply:
my mistaken, resolved.